source/base/fileinputoutput.cpp

```diff
@@ -38,6 +38,7 @@
 
 // C++ variants of standard C header files
 #include <cstdarg>
+#include <cstdio>
```

Comment:
`<cstdio>` is explicitly included here (and in other places) as `std::vsnprintf` is declared there.